### PR TITLE
First step for a generic texture with more than 3 channels, for deep learning applications

### PR DESCRIPTION
**Pull request by francoisruty**

### CODE REVIEW

File: pyredner/material.py [hunk 1]
```diff
@@ -39,19 +40,26 @@ def __init__(self,
             roughness = pyredner.Texture(\
                 torch.tensor([1.0], device = pyredner.get_device()))
 
+        if generic_texture is None:
```
Comment (BachiLi) [inline pyredner/material.py]: We should make this optional. See how I did it with the normal map.

File: pyredner/render_pytorch.py [hunk 2]
```diff
@@ -583,13 +615,19 @@ def backward(ctx,
             d_diffuse_list.append(d_diffuse)
             d_specular_list.append(d_specular)
             d_roughness_list.append(d_roughness)
+            d_generic_list.append(d_generic)
             d_normal_map_list.append(d_normal_map)
             d_diffuse_uv_scale = torch.zeros(2, device = pyredner.get_device())
             d_specular_uv_scale = torch.zeros(2, device = pyredner.get_device())
             d_roughness_uv_scale = torch.zeros(2, device = pyredner.get_device())
+            d_generic_uv_scale = torch.zeros(2, device = pyredner.get_device())
```
Comment (BachiLi) [inline pyredner/render_pytorch.py]: This line is unnecessary since L627 handles it already.

Comment (francoisruty) [reply pyredner/render_pytorch.py]: ok done

**Comment by francoisruty** [comment]
@BachiLi  I made the requested fixes and propagate changes to tensorflow.
Do you prefer to merge this PR and then I make another one to increase the number of channels of generic_texture, or do I keep going on this PR?

**Comment by BachiLi** [comment]
Let's merge it. Thanks a lot for the help!

**Comment by BachiLi** [comment]
FYI I made some changes. Pull from master if you keep working on this.

**Comment by francoisruty** [comment]
all right!

**Pull request merged**